spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -879,6 +879,20 @@ struct ProductionVisitor : public production::Visitor {
         return result;
     }
 
+    /**
+     * Computes how far to advance for `&max-size` parsing where we operate on a limited view,
+     * but do not necessarily consume it fully. The stream for `ncur_max_size` must point
+     * into the `cur` stream.
+     */
+    Expression* computeCurMaxSize(Expression* cur, Expression* ncur_max_size) {
```

Comment:
Nit: This is already in the API of `postParseField`, but isolated like this `ncur_max_size` reads like an expression holding a size where it really is a limited view. Maybe we should call it e.g., `limited` here and elsewhere, and could then call this helper e.g., `advanceLimited` since it has nothing to do with `&max-size` anymore.